airbyte-integrations/connectors/source-stripe/unit_tests/integration/test_events.py

```diff
@@ -0,0 +1,391 @@
+import json
+
+import freezegun
+
+from datetime import datetime, timedelta, timezone
+from typing import Any, Dict, Optional, Tuple
+from unittest import TestCase
+
+from airbyte_cdk.test.catalog_builder import CatalogBuilder
+from airbyte_cdk.test.state_builder import StateBuilder
+from airbyte_cdk.test.entrypoint_wrapper import read, EntrypointOutput
+from airbyte_cdk.test.mock_http import HttpMocker, HttpRequest, HttpResponse
+
+from airbyte_cdk.test.mock_http.request import ANY_QUERY_PARAMS
+from airbyte_cdk.test.mock_http.response_builder import create_builders_from_resource, find_template, FieldPath, HttpResponseBuilder, RecordBuilder
+from airbyte_protocol.models import SyncMode, ConfiguredAirbyteCatalog, AirbyteStreamStatus
+from source_stripe import SourceStripe
+from integration.config import ConfigBuilder
+from integration.pagination import StripePaginationStrategy
+
+
+_STREAM_NAME = "events"
+_NOW = datetime.now(timezone.utc)
+_A_START_DATE = _NOW - timedelta(days=60)
+_ACCOUNT_ID = "account_id"
+_CLIENT_SECRET = "client_secret"
+_AUTHENTICATION_HEADERS = {"Stripe-Account": _ACCOUNT_ID, "Authorization": f"Bearer {_CLIENT_SECRET}"}
+_NO_STATE = {}
+_HTTP_RESPONSE_STATUS_500 = HttpResponse(
+    json.dumps(
+        {"unknown": "maxi297: I could not reproduce the issue hence this response will not look like the actual 500 status response"}
+    ),
+    500
+)
+_AVOIDING_INCLUSIVE_BOUNDARIES = 1
+_SECOND_REQUEST = 1
+_THIRD_REQUEST = 2
+
+def _config() -> ConfigBuilder:
+    return ConfigBuilder().with_account_id(_ACCOUNT_ID).with_client_secret(_CLIENT_SECRET)
+
+
+def _catalog(sync_mode: SyncMode) -> ConfiguredAirbyteCatalog:
+    return CatalogBuilder().with_stream(_STREAM_NAME, sync_mode).build()
+
+
+def _source(catalog: ConfiguredAirbyteCatalog) -> SourceStripe:
+    return SourceStripe(catalog)
+
+
+def _create_builders() -> Tuple[RecordBuilder, HttpResponseBuilder]:
```

Comment:
Could we revisit some of the function names and signatures? For example, I'm not finding it obvious that something called `create_builders_from_resource` or `_create_builders` would return this specific Tuple, and it's causing us to have to do some indexing in `_a_record` and `_a_response` that I don't love. I could understand keeping them coupled if we were using the record & response together, but looking at the usages it doesn't seem like we are.

Reply:
I agree with this. I'll make them separate

Reply:
I added the change in the CDK changes branch (see https://github.com/airbytehq/airbyte/pull/33305/commits/b5b4392b3101a730fe94789aec0365ebb9585752) and updated this branch